x-pack/plugins/ml/public/cases/register_anomaly_charts_attachment.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { EuiDescriptionList } from '@elastic/eui';
+import type { CasesUiSetup } from '@kbn/cases-plugin/public';
+import type { CoreStart } from '@kbn/core/public';
+import type { PersistableStateAttachmentViewProps } from '@kbn/cases-plugin/public/client/attachment_framework/types';
+import { FIELD_FORMAT_IDS } from '@kbn/field-formats-plugin/common';
+import {
+  ANOMALY_EXPLORER_CHARTS_EMBEDDABLE_TYPE,
+  AnomalyChartsEmbeddableInput,
+  getEmbeddableComponent,
+} from '../embeddables';
+import type { MlStartDependencies } from '../plugin';
+import { PLUGIN_ICON } from '../../common/constants/app';
+
+export function registerAnomalyChartsCasesAttachment(
+  cases: CasesUiSetup,
+  coreStart: CoreStart,
+  pluginStart: MlStartDependencies
+) {
+  cases.attachmentFramework.registerPersistableState({
+    id: ANOMALY_EXPLORER_CHARTS_EMBEDDABLE_TYPE,
+    icon: PLUGIN_ICON,
+    displayName: i18n.translate('xpack.ml.cases.anomalyCharts.displayName', {
+      defaultMessage: 'Anomaly charts',
+    }),
+    getAttachmentViewObject: () => ({
+      event: (
+        <FormattedMessage
+          id="xpack.ml.cases.anomalyCharts.embeddableAddedEvent"
+          defaultMessage="added the Anomaly Charts embeddable"
+        />
+      ),
+      timelineAvatar: PLUGIN_ICON,
+      children: React.lazy(() => {
+        return Promise.resolve().then(() => {
+          const EmbeddableComponent = getEmbeddableComponent(
```

Comment:
I think we can move this outside the `children` to avoid creating a new instance of the component on each render.

Reply:
Refactored in https://github.com/elastic/kibana/pull/139628/commits/e712b79bce11f13772dd4b98f73f8d2f0e025f75

Reply:
Still, it will create a new instance of the component each time. You can do:

```
export function registerAnomalySwimLaneCasesAttachment({...}) {
  // remove it from initComponent
  const EmbeddableComponent = getEmbeddableComponent(
    ANOMALY_SWIMLANE_EMBEDDABLE_TYPE,
    coreStart,
    pluginStart
  );
  
  // rest of the code is omitted for brevity
  children: React.lazy(() => {
        return Promise.resolve().then(() => {
          return {
            default: React.memo((props: PersistableStateAttachmentViewProps) => {
              const { persistableStateAttachmentState } = props;

              const dataFormatter = pluginStart.fieldFormats.deserialize({
                id: FIELD_FORMAT_IDS.DATE,
              });

              const inputProps =
                persistableStateAttachmentState as unknown as AnomalyChartsEmbeddableInput;

              return (
                <>
                  <EuiDescriptionList
                    compressed
                    type={'inline'}
                    listItems={[
                      {
                        title: (
                          <FormattedMessage
                            id="xpack.ml.cases.anomalyCharts.description.jobIdsLabel"
                            defaultMessage="Job IDs"
                          />
                        ),
                        description: inputProps.jobIds.join(', '),
                      },
                      {
                        title: (
                          <FormattedMessage
                            id="xpack.ml.cases.anomalyCharts.description.timeRangeLabel"
                            defaultMessage="Time range"
                          />
                        ),
                        description: `${dataFormatter.convert(
                          inputProps.timeRange.from
                        )} - ${dataFormatter.convert(inputProps.timeRange.to)}`,
                      },
                    ]}
                  />
                  // Same instance for all re renders
                  <EmbeddableComponent {...inputProps} />
                </>
              );
            }),
          };
}
```

I am not sure if you can put the rendering code in a separate file and do `await import('./anomaly_charts_attachments');`. Ignore my comment https://github.com/elastic/kibana/pull/139628#discussion_r958361069 if it is not possible.

Reply:
@cnasikas it will only help with having a single `EmbeddableComponent ` instance. Is it possible to preserve the same instance of the entire lazy component here?